src/Contrib/Jaeger/JaegerTransport.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use Jaeger\Thrift\Agent\AgentClient;
+use Jaeger\Thrift\Batch;
+use Jaeger\Thrift\Process;
+use Jaeger\Thrift\Span;
+use Thrift\Exception\TTransportException;
+use Thrift\Protocol\TCompactProtocol;
+
+final class JaegerTransport implements Transport
+{
+
+    // DEFAULT_BUFFER_SIZE indicates the default maximum buffer size, or the size threshold
+    // at which the buffer will be flushed to the agent.
+    const DEFAULT_BUFFER_SIZE = 1;
+
+    private $transport;
+    private $client;
+
+    private $buffer = [];
+    private $process = null;
+    private $maxBufferSize = 0;
+
+    public function __construct($address = '127.0.0.1', $port = 6831, $maxBufferSize = 0)
+    {
+        $this->transport = new ThriftUdpTransport($address, $port);
+        $p = new TCompactProtocol($this->transport);
+        $this->client = new AgentClient($p, $p);
+
+        $this->maxBufferSize = ($maxBufferSize > 0 ? $maxBufferSize : self::DEFAULT_BUFFER_SIZE);
```

Comment:
this could be eliminated by using this const as the constructor parameter default

Reply:
I think I see what you mean. Would that account for negative inputs too? I'm not sure if this class is supposed to just be an implementation detail or consumer facing, but if it's just the former that seems fine (doesn't even appear like that parameter is being used now that I look at it)

Reply:
is a negative buffer size valid? It doesn't sound like it would or should be

Reply:
No clue. But if this is meant to be an internal-only class I can probably just get rid of that parameter and all the associated stuff. Not sure if that's the case though